src/validator.rs

```diff
@@ -157,7 +177,7 @@ fn validate_signed_certificate(
                 &signed_certificate.signature.value,
                 &parent.public_key,
             )?;
-            validate_signed_certificate(signed_parent)?
+            validate_signed_certificate(signed_parent, timestamp)?
```

Comment:
I think it would be best to pass in None here?
There is a check to make sure that the validity period of a parent fully include the validity period of the child. So checking the timestamp in the leaf is enough because if that is valid then the full chain is valid at that moment in time.
The problem is that if we pass in the timestamp here and have chain of certificates A->B->C where both C (leaf) and B (intermediate certificate) will be valid in the future then verification stops at B and the other constraints are not checked between B and C.

Reply:
@MrDarthShoe WDYT?

Reply:
I agree to pass `None` but with comment explaining that timestamp validation only in the leaf is safe (because of validityperiod validation)

Reply:
Changed (with comment).